contracts/examples/bonding-curve/src/bc_function.rs

```diff
@@ -0,0 +1,50 @@
+elrond_wasm::imports!();
+elrond_wasm::derive_imports!();
+
+#[derive(TopEncode, TopDecode, NestedEncode, NestedDecode, TypeAbi, PartialEq, Clone)]
+pub enum SupplyType {
+	Limited,
+	Limitless,
+}
+
+#[derive(TopEncode, TopDecode, NestedEncode, NestedDecode, TypeAbi, PartialEq, Clone)]
+pub struct CurveArguments<BigUint: BigUintApi> {
+	pub supply_type: SupplyType,
+	pub max_supply: BigUint,
+	pub current_supply: BigUint,
+	pub balance: BigUint,
+}
+
+impl<BigUint> CurveArguments<BigUint>
+where
+	BigUint: BigUintApi,
+{
+	fn first_token_available(&self) -> BigUint {
+		self.current_supply.clone() - self.balance.clone()
```

Comment:
No need to clone. You can borrow each value:
&self.current_supply - &self.balance